chia/_tests/wallet/sync/test_wallet_sync.py

```diff
@@ -152,6 +153,74 @@ async def test_request_block_headers_rejected(
+    ph = await wallet.get_new_puzzlehash()
+    for _ in range(2):
+        await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph))
+    await connect_and_get_peer(full_node_api.full_node.server, wallet_server, self_hostname)
+    await full_node_api.wait_for_wallet_synced(wallet_node=wallet_node, timeout=5)
+    # Generate a block with our test spend
+    coins = await wallet_node.wallet_state_manager.coin_store.get_unspent_coins_for_wallet(1)
+    [parent_coin] = [c.coin for c in coins if c.coin.amount == 250_000_000_000]
+    test_spend_amount = 42
+    async with wallet.wallet_state_manager.new_action_scope(DEFAULT_TX_CONFIG, push=True) as action_scope:
+        await wallet.generate_signed_transaction(
+            uint64(test_spend_amount),
+            ph,
+            action_scope,
+            coins={parent_coin},
+            fee=uint64(parent_coin.amount - test_spend_amount),
+        )
+    [tr] = action_scope.side_effects.transactions
+    await full_node_api.wait_transaction_records_entered_mempool(records=[tr])
+    new_block = await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph))
+    # Compute the expected transactions filter
+    [test_spend] = tr.additions
+    byte_array_tx = (
+        [bytearray(test_spend.puzzle_hash)]
+        + [bytearray(coin.puzzle_hash) for coin in new_block.get_included_reward_coins()]
+        + [bytearray(parent_coin.name())]
+    )
+    expected_transactions_filter = bytes(PyBIP158(byte_array_tx).GetEncoded())
+    # Perform the request and check the transactions filter
+    msg = await full_node_api.request_block_headers(
+        wallet_protocol.RequestBlockHeaders(uint32(new_block.height), uint32(new_block.height), True)
+    )
+    assert msg is not None
+    res_block_headers = RespondBlockHeaders.from_bytes(msg.data)
+    block_headers = res_block_headers.header_blocks
+    assert len(block_headers) == 1
+    block_header = block_headers[0]
+    assert block_header.transactions_filter == expected_transactions_filter
+    # Go further and compare this to the outcome of request_header_blocks
+    msg = await full_node_api.request_header_blocks(
+        wallet_protocol.RequestHeaderBlocks(uint32(new_block.height), uint32(new_block.height))
+    )
+    assert msg is not None
+    block_headers_res = RespondBlockHeaders.from_bytes(msg.data)
+    assert block_headers_res.header_blocks == block_headers
+    assert block_headers_res.header_blocks[0].transactions_filter == expected_transactions_filter
+    # Go even further and compare this to the outcome of request_block_header
+    msg = await full_node_api.request_block_header(wallet_protocol.RequestBlockHeader(uint32(new_block.height)))
+    assert msg is not None
+    block_header_res = RespondBlockHeader.from_bytes(msg.data)
+    assert block_header_res.header_block == block_header
+    assert block_header_res.header_block.transactions_filter == expected_transactions_filter
```

Comment:
it seems a bit unnecessarily complicated to involve a wallet. if all you need is a block with transactions, `BlockTools` can do that

Reply:
Sure, done.